### PR TITLE
added support to get notification enable status and take user to settings

### PR DESCRIPTION
**Pull request by sksethi25**
This is for two things
1. Check if user enabled notification or not.
2. Check  show user native popup and take user to settings if he confirm

### CODE REVIEW

File: android/UtilsPlugin.java [hunk 1]
```diff
@@ -241,6 +259,39 @@ public void onNewIntent(Intent gameIntent) {
 		}
 	}
 
+	public void openAppSettings() {
+		Uri packageURI = Uri.parse("package:" + _context.getPackageName());
+		Intent intent = new Intent("android.settings.APPLICATION_DETAILS_SETTINGS", packageURI);
+		intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
```
Comment (bijoshtj) [inline android/UtilsPlugin.java]: what is this flag for??

File: android/UtilsPlugin.java [hunk 2]
```diff
@@ -241,6 +259,39 @@ public void onNewIntent(Intent gameIntent) {
...
+	}
+
+	public void showEnableNotificationPopup(String params) {
+		try {
+			_activity.runOnUiThread(new Runnable() {
+				public void run() {
+					new AlertDialog.Builder(_activity)
+					.setTitle("Enable Notifications")
+					.setMessage("To receive Sudoku of the Day everyday!")
+					.setPositiveButton("Go to Settings", new DialogInterface.OnClickListener() {
```
Comment (rampr) [inline android/UtilsPlugin.java]: It is best if the strings come from the game, so they can be localized

File: ios/UtilsPlugin.mm [hunk 3]
```diff
@@ -143,6 +147,73 @@ - (void)logIt: (NSDictionary *)jsonObject{
 	NSLog(@"{utils-native} LOGIT = %@", message);
 }
 
+- (void)showEnableNotificationPopup: (NSDictionary *)jsonObject{
+    UIAlertController * alert=   [UIAlertController
+                                  alertControllerWithTitle:@"Enable Notifications"
+                                  message:@"To receive Sudoku of the Day everyday!"
```
Comment (rampr) [inline ios/UtilsPlugin.mm]: same as above, strings can come from the game, so it can be localized

File: android/UtilsPlugin.java [hunk 4]
```diff
@@ -241,6 +259,39 @@ public void onNewIntent(Intent gameIntent) {
...
+		_context.startActivity(intent);
+	}
+
+	public void showEnableNotificationPopup(String params) {
+		try {
+			_activity.runOnUiThread(new Runnable() {
+				public void run() {
+					new AlertDialog.Builder(_activity)
+					.setTitle("Enable Notifications")
+					.setMessage("To receive Sudoku of the Day everyday!")
```
Comment (bijoshtj) [inline android/UtilsPlugin.java]: this shouldnt be hard coded. What if we need same functionality for some other notification?

File: android/UtilsPlugin.java (hunk 1 above)
Comment (sksethi25) [inline android/UtilsPlugin.java]: Calling startActivity() from outside of an Activity  context requires the FLAG_ACTIVITY_NEW_TASK flag. Is this really what you want?

got above error and the searched found ,need to add flag

File: js/index.js [hunk 5]
```diff
@@ -80,6 +82,14 @@ exports = new (Class(function () {
         });
     });
 
+    pluginOn('SettingsOpened', function () {
+      invokeCallbacks(cb_notif_enable_popup);
```
Comment (bijoshtj) [inline js/index.js]: wheres the callback/event when user enabled notification from settings??

File: ios/UtilsPlugin.mm (hunk 3 above)
Comment (bijoshtj) [inline ios/UtilsPlugin.mm]: shouldnt be hardcoded

File: ios/UtilsPlugin.mm [hunk 6]
```diff
@@ -143,6 +147,73 @@ - (void)logIt: (NSDictionary *)jsonObject{
 	NSLog(@"{utils-native} LOGIT = %@", message);
 }
 
+- (void)showEnableNotificationPopup: (NSDictionary *)jsonObject{
+    UIAlertController * alert=   [UIAlertController
+                                  alertControllerWithTitle:@"Enable Notifications"
```
Comment (bijoshtj) [inline ios/UtilsPlugin.mm]: not hardcoded

File: ios/UtilsPlugin.mm [hunk 7]
```diff
@@ -143,6 +147,73 @@ - (void)logIt: (NSDictionary *)jsonObject{
...
+                                  message:@"To receive Sudoku of the Day everyday!"
+                                  preferredStyle:UIAlertControllerStyleAlert];
+
+    UIAlertAction* ok = [UIAlertAction
+                         actionWithTitle:@"Go to Settings"
```
Comment (bijoshtj) [inline ios/UtilsPlugin.mm]: same

File: js/index.js (hunk 5 above)
Comment (sksethi25) [inline js/index.js]: this event is when user press open settings btn this is not for enabled settings

File: ios/UtilsPlugin.mm [hunk 8]
```diff
@@ -143,6 +147,73 @@ - (void)logIt: (NSDictionary *)jsonObject{
 	NSLog(@"{utils-native} LOGIT = %@", message);
 }
 
+- (void)showEnableNotificationPopup: (NSDictionary *)jsonObject{
+    UIAlertController * alert=   [UIAlertController
```
Comment (bijoshtj) [inline ios/UtilsPlugin.mm]: space b4 =

File: android/UtilsPlugin.java (hunk 2 above)
Comment (rampr) [inline android/UtilsPlugin.java]: this isn't done

File: js/index.js (hunk 5 above)
Comment (bijoshtj) [inline js/index.js]: i understand this event. dont we need event when user enables?? better to have it

File: android/UtilsPlugin.java [hunk 9]
```diff
@@ -268,12 +268,35 @@ public void openAppSettings() {
 
 	public void showEnableNotificationPopup(String params) {
 		try {
+			String title = "", message = "", open_settings_btn_title = "";
```
Comment (bijoshtj) [inline android/UtilsPlugin.java]: use camelCase. evrywhere else its.

File: android/UtilsPlugin.java [hunk 10]
```diff
@@ -268,12 +268,35 @@ public void openAppSettings() {
...
+			JSONObject ogData = new JSONObject(params);
```
Comment (bijoshtj) [inline android/UtilsPlugin.java]: ogData?? what is itgive meaningful name.

File: android/UtilsPlugin.java [hunk 11]
```diff
@@ -268,12 +268,35 @@ public void openAppSettings() {
...
+				Object o = ogData.get(key);
+				if(key.equals("title")){
+					title = (String) o;
+					continue;
+				}
+				if(key.equals("message")){
+					message = (String) o;
+					continue;
+				}
+				if(key.equals("open_settings_btn_title")){
```
Comment (bijoshtj) [inline android/UtilsPlugin.java]: button_title enough?

File: android/UtilsPlugin.java [hunk 12]
```diff
@@ -268,12 +268,35 @@ public void openAppSettings() {
...
+					open_settings_btn_title = (String) o;
+					continue;
+				}
+			}
+
+			final String ftitle = title;
+			final String fmessage = message;
+			final String fopen_settings_btn_title = open_settings_btn_title;
```
Comment (bijoshtj) [inline android/UtilsPlugin.java]: If final variable is needed, No while loop needed.
String title = data.optString("title", "Title");
similarly for all. So we dont need to use extra 3 variable and can avoid while loop here

File: android/UtilsPlugin.java (hunk 11 above)
Comment (sksethi25) [inline android/UtilsPlugin.java]: no two btns in ios, so no

**Pull request merged**